django/thunderstore/social/api/experimental/views/mod_manager.py

```diff
@@ -0,0 +1,112 @@
+from typing import Sequence, Type
+
+import requests
+import ulid2  # type: ignore
+from django.conf import settings
+from django.contrib.auth import get_user_model, login
+from django.http import HttpResponse
+from drf_yasg.utils import swagger_auto_schema  # type: ignore
+from pydantic import BaseModel
+from rest_framework import serializers
+from rest_framework.exceptions import AuthenticationFailed
+from rest_framework.permissions import BasePermission
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from thunderstore.social.api.experimental.views.complete_login import (
+    get_or_create_auth_user,
+)
+from thunderstore.social.permissions import OauthSharedSecretPermission
+from thunderstore.social.providers import UserInfoSchema
+
+User = get_user_model()
+
+
+class MmLoginRequestBody(serializers.Serializer):
+    jwt = serializers.CharField(label="Authorization token")
+
+
+class MmLoginResponseBody(serializers.Serializer):
+    session_id = serializers.CharField()
+    username = serializers.CharField()
+
+
+class OverwolfProfileSchema(BaseModel):
+    """
+    https://overwolf.github.io/api/profile#get-user-profile-via-token
+    """
+
+    avatar: str
+    nickname: str
+    username: str
+
+
+class ModManagerLoginApiView(APIView):
+    """
+    Login user with information from Overwolf API using the received JWT
+    """
+
+    permission_classes: Sequence[Type[BasePermission]] = [OauthSharedSecretPermission]
+
+    @swagger_auto_schema(
+        operation_id="experimental.auth.mod_manager",
+        request_body=MmLoginRequestBody,
+        responses={200: MmLoginResponseBody()},
+    )
+    def post(self, request: Request) -> HttpResponse:
+        request_data = MmLoginRequestBody(data=request.data)
+        request_data.is_valid(raise_exception=True)
+        jwt: str = request_data.validated_data["jwt"]
+
+        verify_overwolf_jwt(jwt)
+        ow_profile = get_overwolf_user_profile(jwt)
+        user_info = get_user_info(ow_profile)
+        user = get_or_create_auth_user(user_info)
+        login(request, user, "django.contrib.auth.backends.ModelBackend")
+
+        if not request.session.session_key:
+            request.session.create()
+
+        return Response(
+            {
+                "session_id": request.session.session_key,
+                "username": user.username,
+            }
+        )
+
+
+def query_overwolf_jwt_api(path: str, jwt: str) -> requests.Response:
```

Comment:
I feel like this and most of the other overwolf related code should live in the `overwolf_auth` module. I believe you should also be able to use the JWKS validation provided by it with these tokens, which saves one request.

Reply:
`overwolf_auth` contains the SocialAuth/OAuth implementation, and I feel the stuff this PR contains isn't really related to that. I don't think the JWT received from OW API is even signed with the same keys as the OAuth-related JWTs.

I chose this location since the helpers used by "old new" website's authentication (which hopefully are Cyberstorm's helpers in the future) are located here. Although to be fair they are semantically also OAuth stuff and only happen to have helpers that could be reused with the current stuff.